lib/utils/datahub.js

```diff
@@ -53,6 +54,13 @@ class DataHub extends EventEmitter {
     const uploads = resources.map(async resource => {
       // TODO: annoying that the serves parses the s3 url so we have to unparse it!
       const creds = rawstoreUploadCreds[resource.descriptor.path]
+      // Add the path to file in the rawstore - this is use by makeSourceSpec
+      // eslint-disable-next-line camelcase
+      creds.rawstore_url = urljoin(creds.upload_url, creds.upload_query.key)
+      // If a file already is on AWS then just skip uploading for it:
+      if (creds.exists) {
+        return
```

Comment:
Will this `return` terminate the loop? Cause what if one of the files from the resource exists and several others do not? - In this case, it should skip upload for uploaded one and upload rest of them.

Eg: I have a dataset with one file and pushed. After a while, I've got one more resource in the dataset, but the previous one is exactly the same. and I run `data push`. I assume this will push second one only

Reply:
@zelima as discussed on the gitter chat, the `return` doesn't terminate the loop in JS but terminates the current iteration. So everything works as expected.